source/Nevermore.IntegrationTests/Model/LineItem.cs

```diff
@@ -1,14 +1,30 @@
 ﻿using System;
 using Nevermore.Contracts;
+using Octopus.TinyTypes;
 
 namespace Nevermore.IntegrationTests.Model
 {
-    public class LineItem : IDocument
+    public class LineItem : IDocument<LineItemId>, IId
     {
-        public string Id { get; private set; }
+        public LineItemId Id { get; private set; }
         public string Name { get; set; }
         public DateTime PurchaseDate { get; set; }
         public string ProductId { get; set; }
-        public int Quantity { get; set; }
+        public Quantity Quantity { get; set; }
+        string IId.Id => Id?.Value;
+    }
+
+    public class Quantity : TypedInt, Contracts.ITinyType<int>
```

Comment:
@flin-8 This boilerplate will be quite common, i.e. defined once per individual tiny type. If we were to aim to remove the double inheritance, i.e.:
```
public class Quantity : TypedInt
```

...how might we go about that, what implications would that have, and would it be worth it, to simplify this boilerplate?

(my assumption: Nevermore itself - not just its integration tests - would take a dependency on `Octopus.TinyTypes`, and we'd only have one `ITinyType<T>` interface)

Reply:
Mind you, you only need one boilerplate per type:
` public class NMTypedInt : TypedInt, Contracts.ITinyType<int>`
`public class Quantity : NMTypedInt`

This is still not ideal.

The question is are we OK with forcing Nevermore users to take a dependency on `Octopus.TinyTypes`

Reply:
>are we OK with forcing Nevermore users to take a dependency on Octopus.TinyTypes

I believe we discussed this at kick-off and we felt that the costs of breaking this dependency (i.e. requiring every consumer to carefully inherit from both interfaces, whether directly or indirectly) would outweigh the cost (e.g. of potential dependencies on two different versions of Newtonsoft).

Reply:
Correct me if my memory is wrong here @flin-8 :-)